src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -898,6 +898,22 @@ internal static SourceHashAlgorithm TryParseHashAlgorithmName(string arg)
             string baseDirectory,
             SearchOption searchOption,
             IList<Diagnostic> errors)
+        {
+            var sources = ExpandFileNamePatternNonDeterministic(path, baseDirectory, searchOption, errors);
+
+            // Use StringComparer.Ordinal because other comparers may behave
+            // differently depending on the platform.
+            return sources.OrderBy(sf => sf.Path, StringComparer.Ordinal);
```

Comment:
Reference for comment: https://github.com/dotnet/corefx/issues/15825

Does this make sense or is it nuts?